File: src/com/company/Codechef/TomandJerry.cpp
#include <bits/stdc++.h>
using namespace std;
#define lli unsigned long long int

lli p(lli a){
        lli  p = 2;
        while (a%p == 0){
            p = p*2;
        }
        return p;
    }
lli solution(lli ts){
	lli res = 0;
		 if (ts % 2 == 0) {
            lli start = p(ts)/2;
            res = ts/start;
            res/=2;
        } else {
            return ts / 2;
        }
        return res;
	}
int main()
{
	int t;
	cin>>t;
	while(t-- >0){
		lli n;
		cin>>n;
		lli res = solution(n);
		cout<<res<<endl;
	}
	return 0;   
}

File: src/com/company/Codechef/TomAndJerry.java
package com.company;

import java.io.BufferedReader;
import java.io.InputStreamReader;

class TomAndJerry {
    public static void main(String[] args) {
        try {
            BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
            int t = Integer.parseInt(br.readLine().trim());
            while (t-- > 0) {
                long ts = Long.parseLong(br.readLine().trim());
                long res = solution(ts);
                System.out.println(res);
            }
        } catch (Exception e) {
        } finally {
        }
    }

    private static long solution(long ts) {
        if (ts % 2 == 0) {
            long res = 0;
            long start = p(ts)/2;
            long d = ts/start;
            d/=2;
            return d;
        } else {
            return ts / 2;
        }
    }

    static long p(long a) {
        long p = 2;
        while (a % p == 0) {
            p = p * 2;
        }
        return p;
    }
}
